lib/checks/label/duplicate-img-label.js

```diff
@@ -1,18 +1,16 @@
-var imgs = node.querySelectorAll('img');
-var text = axe.commons.text.visible(virtualNode, true).toLowerCase();
-
+const text = axe.commons.text.visible(virtualNode, true).toLowerCase();
 if (text === '') {
 	return false;
 }
 
-for (var i = 0, len = imgs.length; i < len; i++) {
-	var img = imgs[i];
-	var imgAlt = axe.commons.text.accessibleText(img).toLowerCase();
-	if (imgAlt === text &&
-	img.getAttribute('role') !== 'presentation' &&
-	axe.commons.dom.isVisible(img)) {
-		return true;
-	}
-}
+// Get all visible images in the composed tree of the current node
+const images = axe.utils.querySelectorAll(virtualNode, 'img')
+  // Ignore hidden or role=none/presentation images
+  .filter(({ actualNode }) => (axe.commons.dom.isVisible(actualNode) && 
```

Comment:
@dylanb Not sure if this was a bug, or if this is intentional. `isVisible` has a screenreader flag, should this function use that flag? I'm not sure why we care here if the image is positioned off screen or not. Thoughts?